docs/development/contribution.md

```diff
@@ -0,0 +1,22 @@
+
+## Open Issues:
+
+Here's a non-exhaustive list of things that are outright/partially missing from PymoniK that we'd like to see added in: 
+
+- **Unit tests, end-to-end tests** : This project doesn't have any testing associated to it, 0% code coverage. We'd like to change this.
+- **More sophisticated examples** : We'd like to add even more examples and tutorials of common use cases that use ArmoniK under the hood.
+- **PymoniK logger** : The session created/closed/cancelled prints and the different errors should be logged instead of being printed.
+- **Local PymoniK** : Once of the big advantages of the way things have been coded is being able to switch from/to a remote/local context by just removing the `invoke` methods. This could be done even better, by adding a `local=True` flag to PymoniK that makes it so invokes are executed as regular function calls that run locally. The challenge is mainly handling `Pymonik.put`s and the map_invoke. 
+- **Rename `ResultHandle` to `ObjectHandle`** : The naming of `ResultHandle` was choosen because invocations return results, but as it turns out, these results are also then served as inputs. It'd be better for naming (especially since you can put things onto ArmoniK) to rename `ResultHandle` to the more generic `ObjectHandle`.
+- **Cleaner sub-tasking** : Subtasking requires the user to pass in a `delegate=True` flag to invokes, this isn't particularly clean or nice. There must be a better way of doing it. 
+- **Tasks returning multiple results**: As of right now, a task can only return a single result object (even when you return a tuple). There should be support for cases where you'd like to return multiple results from a task and not have them be grouped up into one (multiple smaller objects being passed onto multiple tasks). We think this change would be easier to implement once sub-tasking is in place, since it also involves analyzing what the user will return. There should be pre-execution tests to check if the user is returning a different number of results in different branches of the task and that should result in a failure (invalid task). 
+- **`results.as_completed`** : For more sophisticated and better time-to-execute, we'd like to implement a method for `MultiResultHandle` that allows the user to for instance loop through a `MultiResultHandle` and have the code execute as the result is done/retrieved. Moreover, as a side-effect, having this feature would allow for the usage of `tqdm` to create progress bars which is also really nice. 
+- **Automatic parameter chunking** : There is a size limit on how big ArmoniK objects can be, if the user tries to pass in parameters to a task that is bigger than 30mbs then the code will fail. Ideally, we should be able to detect when we're above the size limit and instead split up our inputs into well sized chunks instead of the user having to do that themselves. 
```

Comment:
There is no limit on ArmoniK object size. What do you mean by that ? Is it a limitation of your implementation ? There are 3 use cases:
- Multiple small results that can be sent in the same request.
- Intermediate size results that are sent in one request.
- Large results that should be splitted and sent through our streamed API.

Reply:
You may want to switch upload method depending on the size of the data.